src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1137,11 +1224,11 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
 
                                 if (op == BinaryOperatorKind.Equal)
                                 {
-                                    this.StateWhenFalse[slot] = true;
+                                    Assign(ref this.StateWhenFalse, slot, true);
```

Comment:
>Assign(ref this.StateWhenFalse, slot, true); [](start = 36, length = 44)

This is not an assignment, state of members shouldn't be affected.

Reply:
Reverted.